plugins/inputs/logfile/logfile_test.go

```diff
@@ -768,7 +773,8 @@ func TestLogsFileRecreate(t *testing.T) {
 	}
 	defer lsrc.Stop()
 
-	for {
+	// Waiting 10 seconds for the recreated temp file to be detected is plenty sufficient on any OS.
+	for start := time.Now(); time.Since(start) < 10 * time.Second; {
```

Comment:
Should we have an assertion after the for-loop to ensure that `len(lsrcs) > 0`? Previously, this would run indefinitely until the test runner timed out. So we would never actually get to the point where we try to access the element in the slice below on line 785:
```
lsrc = lsrcs[0]
```
But now that we're forcing this loop to complete in at most 10 seconds, we can reach a state where there aren't any sources. 

I guess if it's empty and we try to access it, that'll be an indirect failure so it's not like it's uncaught.. but I think we should include an explicit assertion after the loop to be plain and obvious about what we expect.

Reply:
good idea